Sources/Swiftfall/Swiftfall.swift

```diff
@@ -1,635 +1,637 @@
-                let decoded:Set = try decoder.decode(Set.self, from: content)
-                completion(decoded)
-            }
-            catch {
-                // Known Issues:
-                //  * Too broad of a request (needs handler)
-                //
-                // Present an alert if the JSON data cannot be decoded.
-                do {
-                    let decoded:Error = try decoder.decode(Error.self, from: content)
-                    decoded.simplePrint()
-                }
-                catch {
-                    print("Error: \(error)")
-                    completion(nil)
-                }
-                completion(nil)
-            }
-        }
-        task.resume()
-    }
-    
-    public static func getSetList() -> SetList?
-    {
-        let call = "sets/"
-        
-        var setlist: SetList?
-        var stop = false
-        parseSetList(call: call){
-            (newsetlist:SetList?) in
-            setlist = newsetlist
-            stop = true
-        }
-        
-        while(!stop){
-            //Do this until parseCard is done
-        }
-        
-        return setlist
-    }
-    
-    /// Retreives JSON data from URL and parses it with JSON decoder. Thanks Mitchell
-    static func parseSetList(call:String, completion: @escaping (SetList?) -> ()) {
-        
-        let url = URL(string: "\(scryfall)\(call)")
-        let task = URLSession.shared.dataTask(with: url!) {(data, response, error) in
-            
-            guard let content = data else {
-                print("Error: There was no data returned from JSON file.")
-                return
-            }
-            
-            //print("\(String(data: content,encoding: .utf8))")
-            
-            let decoder = JSONDecoder()
-            do {
-                // Decode JSON file starting from Response struct.
-                let decoded:SetList = try decoder.decode(SetList.self, from: content)
-                completion(decoded)
-            }
-            catch {
-                do {
-                    let decoded:Error = try decoder.decode(Error.self, from: content)
-                    decoded.simplePrint()
-                }
-                catch {
-                    print("Error: \(error)")
-                    
-                    completion(nil)
-                }
-                completion(nil)
-            }
-        }
-        task.resume()
-    }
-    
-    public static func getCardList() -> CardList?
-    {
-        let call = "cards/"
-        
-        var cardlist: CardList?
-        var stop = false
-        parseCardList(call: call){
-            (newcardlist:CardList?) in
-            cardlist = newcardlist
-            stop = true
-        }
-        
-        while(!stop){
-            //Do this until parseCard is done
-        }
-        
-        return cardlist
-    }
-    
-    public static func getCardList(page:Int) -> CardList?
```

Comment:
This can also probably get replaced with a call to `synchronouslyGetResource(call:)` if you go that route.